### ISSUE TITLE
Cannot cabal v2-install with non default GHC: Irrefutable pattern failed for pattern Just ghcPkgProg for GHC-8.8

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#5990]
```
 % cabal new-install -w ghc-8.8.1 --lib composition
Warning: Unknown/unsupported 'ghc' version detected (Cabal 3.0.0.0 supports
'ghc' version < 8.8): /opt/ghc/bin/ghc-8.8.1 is version 8.8.0.20190402
Warning: Unknown/unsupported 'ghc' version detected (Cabal 3.0.0.0 supports
'ghc' version < 8.8): /opt/ghc/bin/ghc-8.8.1 is version 8.8.0.20190402
Resolving dependencies...
Build profile: -w ghc-8.8.0.20190402 -O1
In order, the following will be built (use -v for more details):
 - composition-1.0.2.1 (lib) (requires build)
Starting     composition-1.0.2.1 (lib)
Building     composition-1.0.2.1 (lib)
Installing   composition-1.0.2.1 (lib)
Completed    composition-1.0.2.1 (lib)
Distribution/Simple/GHC.hs:1938:5-56: Irrefutable pattern failed for pattern Just ghcPkgProg
```

**Comment by phadej** [comment]
Note that

```
looking for tool ghc-pkg near compiler in /opt/ghc/bin
candidate locations:
["/opt/ghc/bin/ghc-pkg-ghc-8.8.1","/opt/ghc/bin/ghc-pkg-8.8.1","/opt/ghc/bin/ghc-pkg","/opt/ghc/8.8.1/bin/ghc-pkg-ghc-8.8.0.20190402","/opt/ghc/8.8.1/bin/ghc-pkg-8.8.0.20190402","/opt/ghc/8.8.1/bin/ghc-pkg"]
found ghc-pkg in /opt/ghc/bin/ghc-pkg-8.8.1
/opt/ghc/bin/ghc-pkg-8.8.1 --version
/opt/ghc/bin/ghc-pkg-8.8.1 is version 8.8.0.20190402
```

**Comment by phadej** [comment]
Another note: `v2-build` works. It's `v2-install` which is broken.

**Comment by phadej** [comment]
And doesn't need to be weird GHC:

```
% cabal new-install -w ghc-8.4.4 --lib composition                                                                                                     
Resolving dependencies...
Up to date
Distribution/Simple/GHC.hs:1959:5-56: Irrefutable pattern failed for pattern Just ghcPkgProg
```

**Comment by hvr** [comment]
This seems familiar... for convenience, here's the code

https://github.com/haskell/cabal/blob/37bf337317f436e0aee32d4b38a590e790f85fc4/Cabal/Distribution/Simple/GHC.hs#L1959

to state the obvious, it'd be useful to know the value of `ghcPkgProgram` and `progdb` for the succesful (v2-build) case as well as for the unsuccsful (v2-install) case; I suspect this is gonna be easy to fix.

**Comment by maoe** [comment]
This issue seems to be similar to #5941. In #5941 I was using GHC 8.6.4 and FWIW

> Apparently `progdb` in `hcPkgInfo` is empty when the command fails.

**Comment by jrp2014** [comment]
I'm also getting

Distribution/Simple/GHC.hs:1959:5-56: Irrefutable pattern failed for pattern Just ghcPkgProg

with a straight out of the box ubuntu (hvr) cabal-3.0

**Comment by hvr** [comment]
I believe this was fixed by #6195 (which was also included in the 3.0.0.0 release)

**Comment by 23Skidoo** [comment]
Optimistically closing, please reopen if this still happens with 3.0.0.0+.
